src/main/java/com/atlassian/bitbucket/jenkins/internal/deployments/DeploymentPosterImpl.java

```diff
@@ -0,0 +1,88 @@
+import com.atlassian.bitbucket.jenkins.internal.client.exception.BitbucketClientException;
+import com.atlassian.bitbucket.jenkins.internal.config.BitbucketPluginConfiguration;
+import com.atlassian.bitbucket.jenkins.internal.config.BitbucketServerConfiguration;
+import com.atlassian.bitbucket.jenkins.internal.credentials.BitbucketCredentials;
+import com.atlassian.bitbucket.jenkins.internal.credentials.JenkinsToBitbucketCredentials;
+import com.atlassian.bitbucket.jenkins.internal.model.deployment.BitbucketCDCapabilities;
+import com.atlassian.bitbucket.jenkins.internal.model.deployment.BitbucketDeployment;
+import com.cloudbees.plugins.credentials.Credentials;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static java.lang.String.format;
+
+@Singleton
+public class DeploymentPosterImpl implements DeploymentPoster {
+
+    protected static final Logger LOGGER = Logger.getLogger(DeploymentPosterImpl.class.getName());
+
+    private final BitbucketClientFactoryProvider bitbucketClientFactoryProvider;
+    private final JenkinsToBitbucketCredentials jenkinsToBitbucketCredentials;
+    private final BitbucketPluginConfiguration pluginConfiguration;
+
+    @Inject
+    public DeploymentPosterImpl(BitbucketClientFactoryProvider bitbucketClientFactoryProvider,
+                                JenkinsToBitbucketCredentials jenkinsToBitbucketCredentials,
+                                BitbucketPluginConfiguration pluginConfiguration) {
+        this.bitbucketClientFactoryProvider = bitbucketClientFactoryProvider;
+        this.jenkinsToBitbucketCredentials = jenkinsToBitbucketCredentials;
+        this.pluginConfiguration = pluginConfiguration;
+    }
+
+    @Override
+    public void postDeployment(String bbsServerId, String projectKey, String repositorySlug, String revisionSha,
+                               BitbucketDeployment deployment, Run<?, ?> run,
+                               TaskListener taskListener) {
+        Optional<BitbucketServerConfiguration> maybeServer = pluginConfiguration.getServerById(bbsServerId);
+        if (!maybeServer.isPresent()) {
+            taskListener.error(format("Could not send deployment notification to Bitbucket Server: Unknown serverId %s", bbsServerId));
+            return;
+        }
+
+        BitbucketServerConfiguration server = maybeServer.get();
+        Credentials globalAdminCredentials = server.getGlobalCredentialsProvider(run.getParent())
+                .getGlobalAdminCredentials()
+                .orElse(null);
+        BitbucketCredentials credentials = jenkinsToBitbucketCredentials.toBitbucketCredentials(globalAdminCredentials);
+        BitbucketClientFactory clientFactory =
+                bitbucketClientFactoryProvider.getClient(server.getBaseUrl(), credentials);
```

Comment:
Credentials won't be null because `jenkinsToBitbucketCredentials.toBitbucketCredentials` is not marked as `@CheckForNull` which means we can rely on it being non-null